### ISSUE TITLE
Enabling module and public resources

### ISSUE THREAD

**Issue by tzyganu** [magento/magento2#1108]
Right now when I enable a module (didn't test for disabling) using the cli: `php setup/index.php module-enable --modules=Vendor_Module` all the files under `pub/static` are cleared.  
To rebuild them again I need to run again `php dev/tools/Magento/Tools/View/deploy.php`.  
I understand the reasons behind this action but I think it should be optional. If I enable a module that does not change anything in the public resources I may not want to rebuild them. 

I think an extra parameter is needed for the module-enable action.

**Comment by mbrinton01** [comment]
Thank you Marius this is a good idea. We should be able to include this along with some changes the team is working on now. Internal story link: https://jira.corp.x.com/browse/MAGETWO-34621

**Comment by mbrinton01** [comment]
@tzyganu Marius we are considering changing the default behavior so that ``pub/static`` will not be cleared when a module is enabled, and then creating an optional flag to clear the files there.  What are your thoughts on this approach? It is the opposite of what you are asking but would enable you to accomplish the same task.

**Comment by tzyganu** [comment]
@mbrinton01 I don't really care what's the default value of 'clear-files' flag as long as I have the option to delete or not the public files.  Maybe having it set to `false` by default it's better. In case someone forgets to set the value he can delete the files later. but if it's `true` by default there is no way back (except generating them again which takes longer).  
Maybe you can make the flag mandatory so the user must fill in `true` or `false` so it will be clear what happens next.

**Comment by orlangur** [comment]
Why need some additional option and not just [incremental build](http://webpack.github.io/docs/build-performance.html#incremental-builds)?

Pretty naive approach would be "check which resources are bundled within a module and see what's need to be rebuilt" with, probably, some `--force-public-resources-flush` option for really rare cases.

**Comment by mbrinton01** [comment]
Thank you for the feedback. For now we will implement the following:

Files in ``pub/static`` will not be cleared by default on module enable/disable. User will have the option to clear files.

We will consider making this feature more sophisticated in the future to automatically detect what actions should be taken based on the module being enabled/disabled. For now our scope is limited to providing an option to the user.

**Comment by mbrinton01** [comment]
@tzyganu this has now been released in 0.74.0-beta5. The new console is in '\bin'.  Here is the help for 'module:enable' which shows the static files option:

```
$ php magento help module:enable
Usage:
 module:enable [-c|--clear-static-content] [-f|--force] [--all] [--magento_init_params="..."] [module1] ... [moduleN]

Arguments:
 module                      Name of the module

Options:
 --clear-static-content (-c) Clear generated static view files. Necessary, if the module(s) have static view files
 --force (-f)                Bypass dependencies check
 --all                       Enable all modules
 --magento_init_params       Add to any command to customize Magento initialization parameters
                             For example: 'MAGE_MODE=developer&MAGE_DIRS[base][path]=/var/www/example.com&MAGE_DIRS[cache][path]=/var/tmp/cache'
 --help (-h)                 Display this help message
 --quiet (-q)                Do not output any message
 --verbose (-v|vv|vvv)       Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug
 --version (-V)              Display this application version
 --ansi                      Force ANSI output
 --no-ansi                   Disable ANSI output
 --no-interaction (-n)       Do not ask any interactive question
```

The default behavior will not clear files, but the user can choose to do so using the `-c` option.
